asapdiscovery-docking/asapdiscovery/docking/schema/pose_generation.py

```diff
@@ -224,61 +346,98 @@ def _generate_omega_instance(
 
         return omega_generator
 
-    def _generate_poses(
+    def _generate_pose(
         self,
-        prepared_complex: PreppedComplex,
-        ligands: list[Ligand],
+        target_ligand: oechem.OEMol,
+        reference_ligand: oechem.OEMol,
         core_smarts: Optional[str] = None,
-    ) -> tuple[list[oechem.OEMol], list[oechem.OEMol]]:
-        """
-        Use openeye oeomega to generate constrained poses for the input ligands. The core smarts is used to decide
-        which atoms should be constrained if not supplied the MCS will be found by openeye.
+    ) -> oechem.OEMol:
         """
+        Use the configured openeye Omega instance to generate conformers for the target ligand.
 
-        # Make oechem be quiet
-        oechem.OEThrow.SetLevel(oechem.OEErrorLevel_Quiet)
+        Args:
+            target_ligand: The target ligand we want to generate the conformers for.
+            reference_ligand: The ligand which should be used to restrain the target ligand conformers.
+            core_smarts: The smarts which should be used to identify the mcs if not provided it will be determined automatically.
+
+        Returns:
+            The openeye molecule containing the posed conformers.
+        """
 
-        # generate a core fragment used to constrain the generated poses
-        if core_smarts:
+        if core_smarts is not None:
             core_fragment = self._generate_core_fragment(
-                reference_ligand=prepared_complex.ligand, core_smarts=core_smarts
+                reference_ligand=reference_ligand, core_smarts=core_smarts
             )
             use_mcs = False
         else:
             # use the reference ligand and let openeye find the mcs match
-            core_fragment = prepared_complex.ligand.to_oemol()
+            core_fragment = reference_ligand
             use_mcs = True
 
         # build and configure omega
         omega_generator = self._generate_omega_instance(
             core_fragment=core_fragment, use_mcs=use_mcs
         )
 
+        # run omega
+        return_code = omega_generator.Build(target_ligand)
+
+        # deal with strange hydrogen DO NOT REMOVE
+        oechem.OESuppressHydrogens(target_ligand)
```

Comment:
roundtripping hydrogens like this may not be great if we're dealing with specific, pre-specified protonation states on ligands, e.g. when we (somehow) know a protonation state in the binding pocket requires it. Thoughts? Not sure how else to deal with this particular issue though.

Reply:
Good point I did look into this and the function just turns the explicit hydrogens into implicit ones so the parent atoms track them still so the protonation state can not change. This is mainly to just recalculate the positions of the hydrogens as they often get scrambled when doing this kind of conformer generation. The function reference https://docs.eyesopen.com/toolkits/cpp/oechemtk/OEChemFunctions/OESuppressHydrogens.html#OEChem::OESuppressHydrogens